### PR TITLE
Enable and fix errorlint

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Enable the linter and fix the reports.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/oci/runtime_oci.go [hunk 1]
```diff
@@ -226,13 +226,14 @@ func (r *runtimeOCI) CreateContainer(ctx context.Context, c *Container, cgroupPa
 				killErr := cmd.Process.Kill()
 				waitErr := cmd.Wait()
 				if killErr != nil {
-					retErr = fmt.Errorf("failed to kill %+v after failing with: %w", killErr, retErr)
+					retErr = fmt.Errorf("failed to kill %+w after failing with: %w", killErr, retErr)
```
Comment (kwilczynski) [inline internal/oci/runtime_oci.go:229]: This is awkard?

```suggestion
					retErr = fmt.Errorf("failed to kill %w after failing with: %w", killErr, retErr)
```

File: internal/oci/runtime_oci.go [hunk 2]
```diff
@@ -631,13 +633,14 @@ func (r *runtimeOCI) ExecSyncContainer(ctx context.Context, c *Container, comman
...
 				}
 				// Per https://pkg.go.dev/os#ProcessState.ExitCode, the exit code is -1 when the process died because
 				// of a signal. We expect this in this case, as we've just killed it with a signal. Don't append the
 				// error in this case to reduce noise.
-				if exitErr, ok := waitErr.(*exec.ExitError); !ok || exitErr.ExitCode() != -1 {
-					retErr = fmt.Errorf("failed to wait %+v after failing with: %w", waitErr, retErr)
+				var exitErr *exec.ExitError
+				if !errors.As(waitErr, &exitErr) || exitErr.ExitCode() != -1 {
+					retErr = fmt.Errorf("failed to wait %+w after failing with: %w", waitErr, retErr)
```
Comment (kwilczynski) [inline internal/oci/runtime_oci.go:643]:
```suggestion
					retErr = fmt.Errorf("failed to wait %w after failing with: %w", waitErr, retErr)
```

File: pkg/config/config.go [hunk 3]
```diff
@@ -1527,8 +1527,7 @@ func (r *RuntimeHandler) ValidateRuntimePath(name string) error {
 		r.RuntimePath = executable
 		logrus.Debugf("Using runtime executable from $PATH %q", executable)
 	} else if _, err := os.Stat(r.RuntimePath); err != nil && os.IsNotExist(err) {
-		return fmt.Errorf("invalid runtime_path for runtime '%s': %q",
-			name, err)
+		return fmt.Errorf("invalid runtime_path for runtime '%s': %w", name, err)
```
Comment (kwilczynski) [inline pkg/config/config.go:1530]:
```suggestion
		return fmt.Errorf("invalid runtime_path for runtime %q: %w", name, err)
```

File: pkg/config/config.go [hunk 4]
```diff
@@ -1561,8 +1560,7 @@ func (r *RuntimeHandler) ValidateRuntimeConfigPath(name string) error {
 		return errors.New("runtime_config_path can only be used with the 'vm' runtime type")
 	}
 	if _, err := os.Stat(r.RuntimeConfigPath); err != nil && os.IsNotExist(err) {
-		return fmt.Errorf("invalid runtime_config_path for runtime '%s': %q",
-			name, err)
+		return fmt.Errorf("invalid runtime_config_path for runtime '%s': %w", name, err)
```
Comment (kwilczynski) [inline pkg/config/config.go:1563]:
```suggestion
		return fmt.Errorf("invalid runtime_config_path for runtime %q: %w", name, err)
```

File: server/container_create.go [hunk 5]
```diff
@@ -401,7 +401,7 @@ func (s *Server) CreateContainer(ctx context.Context, req *types.CreateContainer
 	if _, err = s.ReserveContainerName(ctr.ID(), ctr.Name()); err != nil {
 		reservedID, getErr := s.ContainerIDForName(ctr.Name())
 		if getErr != nil {
-			return nil, fmt.Errorf("failed to get ID of container with reserved name (%s), after failing to reserve name with %v: %w", ctr.Name(), getErr, getErr)
+			return nil, fmt.Errorf("failed to get ID of container with reserved name (%s), after failing to reserve name with %w: %w", ctr.Name(), getErr, getErr)
```
Comment (kwilczynski) [inline server/container_create.go:404]: Perhaps?

```suggestion
			return nil, fmt.Errorf("failed to get ID of container with reserved name %q after failing to reserve name with %w: %w", ctr.Name(), getErr, getErr)
```

File: server/inspect.go [hunk 6]
```diff
@@ -160,14 +160,14 @@ func (s *Server) GetExtendInterfaceMux(enableProfile bool) *chi.Mux {
 		containerID := chi.URLParam(req, "id")
 		ci, err := s.getContainerInfo(ctx, containerID, s.GetContainer, s.getInfraContainer, s.getSandbox)
 		if err != nil {
-			switch err {
-			case errCtrNotFound:
+			//nolint:gocritic // intentional if else chain
```
Comment (kwilczynski) [inline server/inspect.go:163]: What about switch, still, but without expression?

```go
switch {
case errors.Is(err, errCtrNotFound):
	http.Error(w, "can't find the container with id "+containerID, http.StatusNotFound)
case rrors.Is(err, errCtrStateNil):
	http.Error(w, "can't find container state for container with id "+containerID, http.StatusInternalServerError)
case errors.Is(err, errSandboxNotFound):
	http.Error(w, "can't find the sandbox for container id "+containerID, http.StatusNotFound)
default:
	http.Error(w, err.Error(), http.StatusInternalServerError)
}
```

Comment (saschagrunert) [reply server/inspect.go]: That works, fixed as suggested :+1:

**Comment by saschagrunert** [comment]
/unhold

found the issue :) 

@cri-o/cri-o-maintainers PTAL

**Comment by kwilczynski** [comment]
> found the issue :)

Nice catch!

**Pull request merged**